localstack/aws/spec.py

```diff
@@ -108,41 +112,48 @@ def service_names(self) -> List[ServiceName]:
         return list(self._services.keys())
 
     @cached_property
-    def target_prefix_index(self) -> Dict[str, List[ServiceName]]:
+    def target_prefix_index(self) -> Dict[str, List[ServiceModel]]:
```

Comment:
This is a very good point, the model caching totally slipped my mind.
I changed the index such that it now only uses `ServiceModelIdentifier` (a `NamedTuple`).
This obviously increases the size of the service catalog pickle, but just by ~225kb instead of multiple megabytes: 383,9kb (old) compared to 608,6kb (new)
This results in the following building and loading times of the service catalog:

| Stage | Previous Index | New Index |
| -------- | -------------------- | -------------- |
|Building| 2.0 - 2.1s | 1.8 - 2.2s |
|Loading| 14 - 15ms | 27 - 40ms |

However, the variance was quite high for my tests (on my notebook).
But we can see that the building doesn't really take considerably longer, the loading takes double the time, but is only executed once per LocalStack run and it's an impact of max. 25ms.